### PR TITLE
Added application packages support for Batch

### PR DESCRIPTION
**Pull request by brnleehng**
- Added application packages support with new AutoRest generated
- Flatten Batch update parameters

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@brnleehng__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (brhoan). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by brnleehng** [comment]
Using these swagger specs [https://github.com/Azure/azure-rest-api-specs/blob/master/arm-batch/2015-12-01/swagger/BatchManagement.json](url)

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/Batch/Batch.Tests/Batch.Tests.csproj [hunk 1]
```diff
@@ -27,4 +28,48 @@
       <Name>BatchManagement</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <None Include="packages.config" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.Azure.Common, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>$(LibraryNugetPackageFolder)\Microsoft.Azure.Common.2.0.4\lib\net45\Microsoft.Azure.Common.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManagement/Batch/Batch.Tests/Batch.Tests.csproj]: please use latest version - 2.1.0

File: src/ResourceManagement/Batch/BatchManagement/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@
 [assembly: AssemblyDescription("Provides management functions for Microsoft Azure Batch services.")]
 
 [assembly: AssemblyVersion("1.0.0.0")]
```
Comment (hovsepm) [inline src/ResourceManagement/Batch/BatchManagement/Properties/AssemblyInfo.cs]: change this to 2.0.0.0

**Pull request merged**